pkg/genericcli/context.go

```diff
@@ -0,0 +1,682 @@
+	}
+
+	wrapper := &cliWrapper{
+		cfg: c,
+	}
+
+	cmd := NewCmds(&CmdsConfig[
+		*Context,
+		*contextUpdateRequest,
+		*Context,
+	]{
+		GenericCLI:      NewGenericCLI(wrapper),
+		BinaryName:      c.BinaryName,
+		Singular:        "context",
+		Plural:          "contexts",
+		Description:     "Manage CLI contexts. A context defines the connection properties (API URL, token, etc.) for a backend. Use \"-\" to switch to the previously used context.",
+		Aliases:         []string{"ctx"},
+		Args:            []string{KeyName},
+		Sorter:          contextSorter(),
+		DescribePrinter: c.DescribePrinter,
+		ListPrinter:     func() printers.Printer { return newPrinterFromCLI(c) },
+		In:              c.In,
+		Out:             c.Out,
+		OnlyCmds: OnlyCmds(
+			DescribeCmd,
+			ListCmd,
+			CreateCmd,
+			UpdateCmd,
+			DeleteCmd,
+		),
+		RootCmdMutateFn: func(cmd *cobra.Command) {
+			cmd.Args = cobra.MaximumNArgs(1)
+			cmd.RunE = func(cmd *cobra.Command, args []string) error {
+				// '$ BinaryName context' (no args) should be equal to '$ BinaryName context list'
+				if len(args) == 0 {
+					listCmd, _, err := cmd.Find([]string{"list"})
+					if err != nil {
+						return fmt.Errorf("internal: list command not found: %w", err)
+					}
+					return listCmd.RunE(listCmd, []string{})
+				}
+
+				// '$ BinaryName context -' or '$ BinaryName context <name>' should behave like 'switch'
+				if len(args) == 1 {
+					return c.setContext(args)
+				}
+
+				// Probably too many args, fallback to help
+				return fmt.Errorf("too many arguments")
+			}
+		},
+		DescribeCmdMutateFn: func(cmd *cobra.Command) {
+			cmd.Args = cobra.MaximumNArgs(1)
+
+			originalRunE := cmd.RunE
+
+			cmd.RunE = func(cmd *cobra.Command, args []string) error {
+				// If no args are provided, try to use the current context
+				if len(args) == 0 {
+					ctxs, err := c.GetContexts()
+					if err != nil {
+						return fmt.Errorf("unable to get contexts to determine current: %w", err)
+					}
+					if ctxs.CurrentContext == "" {
+						return fmt.Errorf("no context name provided and no context is currently active")
+					}
+					args = []string{ctxs.CurrentContext}
+				}
+				return originalRunE(cmd, args)
+			}
+		},
+		ListCmdMutateFn: func(cmd *cobra.Command) {
+			cmd.Args = cobra.ExactArgs(0)
+		},
+		CreateCmdMutateFn: func(cmd *cobra.Command) {
+			cmd.Flags().String(KeyName, "", "set the name of the context")
+			cmd.Flags().String(KeyAPIURL, "", "set the api-url for this context")
+			cmd.Flags().String(KeyAPIToken, "", "set the api-token for this context")
+			cmd.Flags().String(KeyDefaultProject, "", "set a default project to operate on")
+			cmd.Flags().Duration(KeyTimeout, 0, "set a default request timeout")
+			cmd.Flags().Bool(KeyActivate, false, "immediately switches to the new context")
+			cmd.Flags().String(KeyProvider, "", "set the login provider for this context")
+
+			Must(cmd.MarkFlagRequired(KeyName))
+			Must(cmd.MarkFlagRequired(KeyAPIToken))
```

Comment:
The problem marking the flags required for creation and update commands that the commands also have an added `--file` flag (creation / update through file). So it's actually mutually exclusive. Usually requiring this flag is checked in the `CreateRequestFromCLI` functions. 

You can for now keep it but you will get to this when you have more tests.